### PR TITLE
Add tooltips to the action bar

### LINKED ISSUE #5933: A help tooltip when hovering the icons near the node.
A tooltip with brief information about the icon should be displayed for:
1. Output context switch icon
2. Visualization switch icon

### QA Acceptance Criteria

```
GIVEN an open project or demo scene
WHEN I hover the visualization switch icon to the right of the node
THEN A tooltip with a short description of the button is displayed
THEN A tooltip disappears after a short delay

WHEN I hover the output context switch icon to the right of the node
THEN A tooltip with a short description of the button is displayed
THEN A tooltip disappears after a short delay
```

### PR DESCRIPTION
**Pull request by Procrat**
### Pull Request Description

Implements #5933: adding tooltips to the buttons next to nodes.

To make the UI consistent, I've added tooltips to the `ToggleButton` class directly, since whenever you have an icon button, it seems helpful to have a tooltip.

`ToggleButton` is only used for the profiling button in the top-right corner and the buttons next to nodes. The output context switch button [isn't implemented yet](https://github.com/enso-org/enso/issues/5929), but once it is, adding a tooltip should be one-liner.

Recording 2023-03-22 at 17 21 58

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: app/gui/view/graph-editor/src/component/node/action_bar.rs [hunk 1]
```diff
@@ -89,11 +93,12 @@ struct Icons {
 }
 
 impl Icons {
-    fn new() -> Self {
+    fn new(app: &Application) -> Self {
         let display_object = display::object::Instance::new();
-        let freeze = ToggleButton::new();
-        let visibility = ToggleButton::new();
-        let skip = ToggleButton::new();
+        let freeze = ToggleButton::new(app, tooltip::Style::set_label("Freeze".to_owned()));
```
Comment (vitvakatu) [inline app/gui/view/graph-editor/src/component/node/action_bar.rs]: Please extract string literals into constants

File: CHANGELOG.md [hunk 2]
```diff
@@ -126,6 +126,7 @@
   shortened labels for entries with long module paths. When an option is
   selected from the dropdown, the necessary module imports are inserted,
   eliminating the need for fully qualified names.
+- [Icon buttons now have tooltips.][6035]
```
Comment (wdanilo) [inline CHANGELOG.md]: Please write a better description - it is user-facing changelog.

File: app/gui/view/graph-editor/src/component/node/action_bar.rs [hunk 3]
```diff
@@ -89,11 +93,12 @@ struct Icons {
...
+        let visibility =
```
Comment (wdanilo) [inline app/gui/view/graph-editor/src/component/node/action_bar.rs]: please follow style guide-  make express one-liners.

File: app/gui/view/graph-editor/src/component/node/action_bar.rs [hunk 4]
```diff
@@ -324,3 +329,34 @@ impl display::Object for ActionBar {
         self.model.display_object()
     }
 }
+
+
+#[cfg(test)]
+mod test {
```
Comment (wdanilo) [inline app/gui/view/graph-editor/src/component/node/action_bar.rs:347]: section missing

File: lib/rust/ensogl/component/toggle-button/src/lib.rs [hunk 5]
```diff
@@ -241,16 +244,16 @@ impl<Shape: ColorableShape + 'static> ToggleButton<Shape> {
 
             frp.source.mouse_over <+ icon.mouse_over;
             frp.source.mouse_out  <+ icon.mouse_out;
+            frp.source.is_hovered <+ bool(&icon.mouse_out,&icon.mouse_over);
+            frp.source.is_pressed <+ bool(&icon.mouse_up_primary,&icon.mouse_down_primary);
```
Comment (wdanilo) [inline lib/rust/ensogl/component/toggle-button/src/lib.rs]: there should be space after `,` - format it like rustfmt pls

File: lib/rust/ensogl/component/toggle-button/src/lib.rs [hunk 6]
```diff
@@ -202,27 +204,28 @@ pub struct ToggleButton<S: Shape> {
...
-    fn init_frp(self) -> Self {
+    fn init_frp(self, app: &Application, tooltip_style: tooltip::Style) -> Self {
         let network = &self.frp.network;
         let frp = &self.frp;
         let model = &self.model;
         let color = color::Animation::new(network);
         let icon = &model.icon.events;
 
+        // Explicitly define the tooltip placement if the default was used, to ensure that, when
+        // multiple tooltips are being used, we always override to the one we want.
```
Comment (wdanilo) [inline lib/rust/ensogl/component/toggle-button/src/lib.rs]: I don't understand this sentence. What is "the one that we want"? Also, it has some minor English grammar mistakes. See granary or chatgpt to check correctness, please.

**Review by wdanilo — CHANGES_REQUESTED** [review]
Beside the code comments:
1. The text on the tooltip is not vertically centered - it looks bad. This should be fixed.
2. The tooltip should appear with some delay. This should be controlled by the mouse tooltip logic. We can modify it in a separate task, but its important to fix it.

File: CHANGELOG.md (hunk 2 above)
Comment (Procrat) [reply CHANGELOG.md]: Any suggestions? Please be constructive when providing PR comments.

**Comment by Procrat** [comment]
@wdanilo: I agree on both points you raise. However, they seem orthogonal to this change, so I suggest doing both in a separate PR.

File: lib/rust/ensogl/component/toggle-button/src/lib.rs (hunk 6 above)
Comment (Procrat) [reply lib/rust/ensogl/component/toggle-button/src/lib.rs]: I've added some extra context. Let me know if you're still confused. Happy to jump into a call to explain it.

**Comment by vitvakatu** [comment]
@Procrat please create an issue for text position + show delay, let's make it now, in a separate PR.

**Comment by Procrat** [comment]
I've create an issue [here](https://github.com/enso-org/enso/issues/6059) for the visual improvement of tooltips in general.

File: CHANGELOG.md (hunk 2 above)
Comment (wdanilo) [reply CHANGELOG.md]: I was sure it was constructive enough. User-facing changelog should describe what is the change, why it was made and what benefits it gives users.

Comment (vitvakatu) [reply CHANGELOG.md]: I played with GPT for a minute:
```
[Added tooltips to icon buttons][6035] for improved usability. Users can now quickly understand each button's function.
```

**Review by wdanilo — APPROVED** [review]

**Comment by MichaelMauderer** [comment]
QA: Passed.

**Pull request merged**